File: backend/voting-app/src/main/java/com/tu/votingapp/dto/general/DocumentDTO.java
package com.tu.votingapp.dto.general;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class DocumentDTO {
    private Long id;
    private String number;
    private Date validFrom;
    private Date validTo;
    private String issuer;
    private Integer gender;
    private Date dateOfBirth;
    private String permanentAddress;
    private UserDTO user;
}

File: frontend/android/app/src/main/java/com/example/android/dto/response/DocumentDTO.kt
package com.example.android.dto.response

import kotlinx.serialization.Serializable

@Serializable
data class DocumentDTO(
    val id: Long? = null,
    val number: String,
    val validFrom: String,        // ISO date format: "yyyy-MM-dd"
    val validTo: String,          // ISO date format: "yyyy-MM-dd"
    val issuer: String,
    val gender: Int,
    val dateOfBirth: String,      // ISO date format: "yyyy-MM-dd"
    val permanentAddress: String,
    val user: UserDTO? = null     // optional to avoid recursion issues in serialization
)